contrib/init/systemd/docker.service

```diff
@@ -5,8 +5,9 @@ After=network.target docker.socket
 Requires=docker.socket
 
 [Service]
-ExecStart=/usr/bin/docker -d -H fd://
+ExecStart=/usr/bin/docker daemon -H fd://
 MountFlags=slave
+Restart=on-failure
```

Comment:
Perhaps is something good, but if not directly related, it should be a separate PR